### PR TITLE
chore: prepare version 0.5.0

### PR DESCRIPTION
**Pull request by soulcramer**
## 📋 Changes description

Upgrade the version fro m0.4.2 to 0.5.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,37 @@
...
+
+_2023-09-20_
+
+### Spark
+
+* 🆕 Add IconButtons with all intents, shapes and sizes
+* 🆕 Add loading state to the IconButton component
+* 🆕 Add icon toggle button
+* 💄 Use M2 elevation system instead of the M3 one
+* ⬆️[Bump `compileSdk` and `targetSdk` to 34
```
Comment (SimonMarquis) [inline CHANGELOG.md:18]:
```suggestion
* ⬆️ Bump `compileSdk` and `targetSdk` to 34
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,37 @@
...
 
+## [0.5.0]
+
+_2023-09-20_
+
+### Spark
+
+* 🆕 Add IconButtons with all intents, shapes and sizes
+* 🆕 Add loading state to the IconButton component
+* 🆕 Add icon toggle button
```
Comment (SimonMarquis) [inline CHANGELOG.md:16]: nit: should we also use the proper composable name here? (or is it something else)
```suggestion
* 🆕 Add IconToggleButton
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,37 @@
...
+* 💄 Use M2 elevation system instead of the M3 one
+* ⬆️[Bump `compileSdk` and `targetSdk` to 34
+* 🎨 Change colors from LBC & KA
+* 🐛 Make the readonly Textfields not take the focus look when focused
+* 📝 Replace onliner with two distinct commands
```
Comment (SimonMarquis) [inline CHANGELOG.md:21]: Sorry for the typo comming from my commit message 🙈
```suggestion
* 📝 Replace oneliner with two distinct commands
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,37 @@
 ## [Unreleased]
 
 
+## [0.5.0]
+
+_2023-09-20_
+
+### Spark
```
Comment (SimonMarquis) [inline CHANGELOG.md:12]: nit: for consistency, could we always use backticks when referring to class names?

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,37 @@
...
+* 📝 Replace onliner with two distinct commands
+* 🐛 ModalFullScreenScaffold top padding issue when no illustration
+
+
+### Catalog App
+
+* 🆕 Add Icons demo to CatalogApp
+* 🆕 Add examples for tab component
+* 💄 Add Kleinanzeigen theme
+* 🎨 Add illustrations for components item
```
Comment (SimonMarquis) [inline CHANGELOG.md:30]: Not trying to be too pushy, but are these illustrations being validated, or at least tracked somewhere to be updated later?

Comment (soulcramer) [reply CHANGELOG.md:30]: Ah this was tracked in https://github.com/adevinta/spark-android/issues/714 but in the daily they were okay with these. At least they don't plan/have time to produce illustrations for each component for each demo apps

**Review by kazaky — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (SimonMarquis) [reply CHANGELOG.md:30]: ACK I was not aware of this 👍

**Pull request merged**